google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -356,36 +311,34 @@ def _auth_init(self, credentials: Optional[Credentials]) -> None:
 
         self._credentials = credentials
 
-    async def _force_refresh(self) -> bool:
+    def force_refresh(self) -> bool:
+        """
+        Forces a new refresh attempt and returns a boolean value that indicates
+        whether the attempt was successful.
+        """
+        # if a new refresh is already in progress, then block on the result
         if self._refresh_in_progress.is_set():
-            # if a new refresh is already in progress, then block on the result
-            self._current = await self._next
+            self._current = self._next
             return True
+        # if next refresh hasn't started, cancel it and schedule new one immediately
         try:
             self._next.cancel()
             # schedule a refresh immediately with no delay
-            self._next = self._loop.create_task(self._schedule_refresh(0))
-            self._current = await self._next
+            self._next = self._schedule_refresh(0)
+            self._current = self._next
             return True
         except Exception as e:
             # if anything else goes wrong, log the error and return false
             logger.exception("Error occurred during force refresh attempt", exc_info=e)
             return False
```

Comment:
This section is feeling a bit wonky. Where do we expect an `Exception` to be thrown? Is it just `cancel()`? If so I think we should move L327-L329 out of the try/catch. I don't think we want to return false if we actually scheduled a refresh (but I don't think we can get an error there anymore)